File: 1537-maximum-score-after-splitting-a-string/solution.py
class Solution(object):
    def maxScore(self, s):
        """
        :type s: str
        :rtype: int
        """
        totalOnes = 0
        for i in xrange(len(s)):
            bit = s[i]
            if bit == '1':
                totalOnes += 1
        
        maxScore = float('-inf')
        onesCounted = 0
        zerosCounted = 0
        for i in xrange(len(s)-1):
            bit = s[i]
            if bit == '1':
                onesCounted += 1
            else:
                zerosCounted += 1
            score = zerosCounted + (totalOnes - onesCounted)
            maxScore = max(maxScore, score)
    
        return maxScore
        

File: 1537-maximum-score-after-splitting-a-string/solution.cpp
class Solution {
public:
    int maxScore(string s) {
        int totalOnes = 0;
        for (size_t i = 0; i < s.size(); ++i) {
            if (s[i] == '1') totalOnes++;
        }
        int numOnesSeen = 0;
        int maxScore = 0;
        for (size_t i = 0; i < s.size() - 1; ++i) {
            if (s[i] == '1') numOnesSeen++;
            int subStringLen = (i + 1);

            int numZerosLeft = subStringLen - numOnesSeen;
            int numOnesRight = totalOnes - numOnesSeen;
            
            int score = numZerosLeft + numOnesRight;
            
            maxScore = max(maxScore, score);
        }

        return maxScore;
    }
};
